README.md

```diff
@@ -12,8 +17,74 @@ you're free to choose what technology stack to run on, but on top of that, you c
 the exact same components wherever you need them. For example, you can take a component
 originally developed in a Jupyter Notebook and embed it in your production application
 without changing anything about the component itself.
+<!---Documatic-section-group: arch-start--->
+
+
+<!---Documatic-section-group: arch-end--->
+
+<!---Documatic-section-group: helloworld-start--->
+
+
+## Code Overview
+
+The codebase has a 2-deep folder structure, with 31 in total.
+<!---Documatic-section-helloworld: setup-start--->
+The codebase is compatible with Python 3.10 and above, because of pipe type hint in `src/idom/testing.py`.
```

Comment:
This is not strictly true. I'm using `from __future__ import annotations` which delays type annotation evaluation: [PEP563](https://www.python.org/dev/peps/pep-0563/). However that doesn't necessarily mean everything else will work in 3.10